From 8868a3a4c7bfaa356d8c0eb7c6abb8b980f63d99 Mon Sep 17 00:00:00 2001
From: Joey Hess <joeyh@joeyh.name>
Date: Thu, 1 Apr 2021 12:19:47 -0400
Subject: Fix build with persistent-2.12.0.1

persistent stopped using askLogFunc, and the thing to use is askLoggerIO
from monad-logger. Bumped the dep to the first version that contained that.

Note that the i386ancient build uses a newer monad-logger than 0.3.10,
so the new versioned dep should not break it, and presumably nothing else
either.

This commit was sponsored by Noam Kremen on Patreon.
---
 CHANGELOG                                                  |  6 ++++++
 Database/Handle.hs                                         |  6 +++---
 doc/bugs/git-annex_8.20210330_fails_to_build_in_brew.mdwn  |  2 ++
 .../comment_1_ddf94a16e7075c218473b01b15fdca3d._comment    | 14 ++++++++++++++
 git-annex.cabal                                            |  2 +-
 5 files changed, 26 insertions(+), 4 deletions(-)
 create mode 100644 doc/bugs/git-annex_8.20210330_fails_to_build_in_brew/comment_1_ddf94a16e7075c218473b01b15fdca3d._comment

diff --git a/CHANGELOG b/CHANGELOG
index 6f0e0bf35..33c80fa5b 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,9 @@
+git-annex (8.20210331) UNRELEASED; urgency=medium
+
+  * Fix build with persistent-2.12.0.1
+
+ -- Joey Hess <id@joeyh.name>  Thu, 01 Apr 2021 12:17:26 -0400
+
 git-annex (8.20210330) upstream; urgency=medium
 
   * Behavior change: When autoenabling special remotes of type S3, weddav,
diff --git a/Database/Handle.hs b/Database/Handle.hs
index 2109ad9ac..d7f1822dc 100644
--- a/Database/Handle.hs
+++ b/Database/Handle.hs
@@ -26,7 +26,7 @@ import qualified Database.Sqlite as Sqlite
 import Control.Monad
 import Control.Monad.IO.Class (liftIO)
 import Control.Monad.IO.Unlift (MonadUnliftIO, withRunInIO)
-import Control.Monad.Logger (MonadLogger)
+import Control.Monad.Logger (MonadLoggerIO, askLoggerIO)
 import Control.Concurrent
 import Control.Concurrent.Async
 import Control.Exception (throwIO, BlockedIndefinitelyOnMVar(..))
@@ -245,7 +245,7 @@ runSqliteRobustly tablename db a = do
 -- Like withSqlConn, but more robust.
 withSqlConnRobustly
 	:: (MonadUnliftIO m
-		, MonadLogger m
+		, MonadLoggerIO m
 		, IsPersistBackend backend
 		, BaseBackend backend ~ SqlBackend
 		, BackendCompatible SqlBackend backend
@@ -254,7 +254,7 @@ withSqlConnRobustly
 	-> (backend -> m a)
 	-> m a
 withSqlConnRobustly open f = do
-	logFunc <- askLogFunc
+	logFunc <- askLoggerIO
 	withRunInIO $ \run -> bracket
 		(open logFunc)
 		closeRobustly
